pkg/http.ts

```diff
@@ -202,6 +223,11 @@ export class HttpClient implements Requester {
       backend: this.options?.backend,
     };
 
+    if (this.readYourWrites) {
+      const newHeader = this.upstashSyncToken;
+      this.headers["upstash-sync-token"] = newHeader;
+    }
+
```

Comment:
Can we omit this and do header init in the if below? This a bit hard to follow. Felt unintentional

Reply:
There are two steps here,
* For each request, we set the header from the header of the previous requests' response, which is set as `upstashSyncToken` after the fetch request.
* After the fetch request, we get the response header and save it for the next object.

Since we don't have a global header object, we can't update the header in the second if, since we'd have already sent the request. And we also don't have a global header config, and we arrange that per request.

Reply:
Maybe add a small comment documenting this.